service/dap/server.go

```diff
@@ -524,6 +527,12 @@ func (s *Session) ServeDAPCodec() {
 			s.config.triggerServerStop()
 		}
 	}()
+
+	s.disconnectCh = make(chan struct{})
+	defer func() {
```

Comment:
Can just `defer close(s.disconnectCh)` without the anonymous function.

Reply:
Actually, maybe the channel should live on the `s.conn` connection and be closed via the above `defer s.conn.Close()`?

Reply:
Agree, didn't realise s.conn is a wrapper of `net.Conn`